### PR TITLE
chore(evm): add comment regarding RANDOM opcode

### PR DESCRIPTION
**Pull request by austinchandra**
## Description

Ref: #1512 

Go-Ethereum [sets the random field](https://github.com/ethereum/go-ethereum/blob/1b8a392153b39fbbde17536c730d96510e57341f/core/evm.go#L56) using the block header's mixDigest, which is [defined](https://eips.ethereum.org/EIPS/eip-4399#block-structure) to be a hash corresponding to a random hash generated by the previous block.

We can mirror this by using the consensus hash from each Cosmos block.

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by austinchandra** [comment]
Not sure if there are any crucial differences between the ranges of Cosmos's consensus hash and Ethereum's mixHash/mixDigest. It's possible they have a different range of output, which could change behavior.

File: x/evm/keeper/state_transition.go [hunk 1]
```diff
@@ -79,6 +79,9 @@ func (k *Keeper) NewEVM(
 	tracer vm.EVMLogger,
 	stateDB vm.StateDB,
 ) evm.EVM {
+	// Use BlockHash for random field
+	random := common.BytesToHash(ctx.HeaderHash())
```
Comment (fedekunze) [inline x/evm/keeper/state_transition.go:83]: looks like the random number generated corresponds to the [RANDAO](https://eth2.incessant.ink/book/06__building-blocks/02__randomness.html#enter-randao) and not the block header. 

The header hash is not random by definition

**Review by fedekunze — CHANGES_REQUESTED** [review]
See comment about using RANDAO

File: x/evm/keeper/state_transition.go (hunk 1 above)
Comment (fedekunze) [reply x/evm/keeper/state_transition.go:83]: @austinchandra if we can't add RANDAO (most likely blocked on ABCI++ VoteExtension), I'd suggest adding a note saying that the RANDOM opcode is currently not supported.

Comment (austinchandra) [reply x/evm/keeper/state_transition.go:83]: Do we have a doc page listing unsupported features vs. Ethereum?

**Review by fedekunze — APPROVED** [review]

**Review by facs95 — APPROVED** [review]

**Pull request merged**